catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/ResourceOperations.java

```diff
@@ -0,0 +1,750 @@
+
+    private static final String DEFAULT_RESOURCE_NOT_FOUND_MESSAGE = "Unknown resource request";
+
+    private Supplier<CatalogProvider> catalogSupplier;
+
+    // Inject properties
+    private FrameworkProperties frameworkProperties;
+
+    private QueryOperations queryOperations;
+
+    private OperationsSecuritySupport opsSecuritySupport;
+
+    public ResourceOperations(FrameworkProperties frameworkProperties,
+            QueryOperations queryOperations, OperationsSecuritySupport opsSecuritySupport) {
+        this.frameworkProperties = frameworkProperties;
+        this.queryOperations = queryOperations;
+        this.opsSecuritySupport = opsSecuritySupport;
+
+        setId(SystemInfo.getSiteName());
+        setVersion(SystemInfo.getVersion());
+        setOrganization(SystemInfo.getOrganization());
+    }
+
+    void setCatalogSupplier(Supplier<CatalogProvider> catalogSupplier) {
+        this.catalogSupplier = catalogSupplier;
+    }
+
+    //
+    // Delegate methods
+    //
+    ResourceResponse getEnterpriseResource(ResourceRequest request, boolean fanoutEnabled)
+            throws IOException, ResourceNotFoundException, ResourceNotSupportedException {
+        String methodName = "getEnterpriseResource";
+        LOGGER.debug("ENTERING: {}", methodName);
+        ResourceResponse resourceResponse = getResource(request, true, null, fanoutEnabled);
+        LOGGER.debug("EXITING: {}", methodName);
+        return resourceResponse;
+    }
+
+    Map<String, Set<String>> getEnterpriseResourceOptions(String metacardId,
+            boolean fanoutEnabled) throws ResourceNotFoundException {
+        LOGGER.trace("ENTERING: getEnterpriseResourceOptions");
+        Set<String> supportedOptions = Collections.emptySet();
+
+        try {
+            QueryRequest queryRequest = new QueryRequestImpl(createMetacardIdQuery(metacardId),
+                    true,
+                    null,
+                    null);
+            QueryResponse queryResponse = queryOperations.query(queryRequest,
+                    null,
+                    false,
+                    fanoutEnabled);
+            List<Result> results = queryResponse.getResults();
+
+            if (!results.isEmpty()) {
+                Metacard metacard = results.get(0)
+                        .getMetacard();
+                String sourceIdOfResult = metacard.getSourceId();
+
+                if (sourceIdOfResult != null && sourceIdOfResult.equals(getId())) {
+                    // found entry on local source
+                    supportedOptions = getOptionsFromLocalProvider(metacard);
+                } else if (sourceIdOfResult != null && !sourceIdOfResult.equals(getId())) {
+                    // found entry on federated source
+                    supportedOptions = getOptionsFromFederatedSource(metacard, sourceIdOfResult);
+                }
+            } else {
+                String message = "Unable to find metacard " + metacardId + " on enterprise.";
```

Comment:
I'm assuming you are tracking these, (maybe on the next round of the refactoring) to use something like `String.format()`.